07a_system_activity_plots.R

```diff
@@ -189,75 +190,139 @@ renderSystemPlot <- function(id) {
-      xend = if_else(group.id == last(group.id), last(group.id), group.id + 1),
-      y = yend,
-      yend = yend
-    ),
-    linewidth = .3,
-    colour = "gray25",
-    linetype = "dashed",
-    show.legend = FALSE,
-    inherit.aes = FALSE
-  ) +
-  ggrepel::geom_text_repel(# the labels
-    aes(
-      x = group.id,
-      label = paste0(scales::comma(abs(values))),
-      y = rowSums(cbind(ystart, values / 2)),
-      segment.colour = "gray33"
-    ),
-    nudge_x = -.5,
-    arrow = arrow(type = "open", length = unit(.1, "inches")),
-    colour = "#4e4d47",
-    # alpha = .7,
-    size = 5,
-    inherit.aes = FALSE
-  ) +
-  scale_fill_manual(values = colors) + # color palette
-  scale_y_continuous(expand = c(0,0)) + # distance between bars and x axis line
-  scale_x_continuous(labels = str_wrap(df$Time %>% unique(), width = 10), # x axis labels
-                   breaks = df$group.id %>% unique()) +
-  theme_void() + # totally clear all theme elements
-  theme(# add back in what theme elements we want
-    text = element_text(size = 16, colour = "#4e4d47"),
-    axis.text.x = element_text(size = 16),
-    axis.ticks.x = element_line(),
-    axis.line.x = element_line(colour = "#4e4d47", linewidth = 0.5),
-    axis.ticks.length.x = unit(.15, "cm"),
-    plot.margin = unit(c(1, 1, 1, 1), "lines"),
-    legend.text = element_text(size = 16),
-    legend.title = element_blank()#,
-    # legend.position = "none"
-  )
+    ggplot(df, aes(x = group.id, fill = Status)) +
+      geom_rect(
+        # the bars
+        aes(
+          xmin = group.id - 0.25,
+          # control bar gap width
+          xmax = group.id + 0.25,
+          ymin = ystart,
+          ymax = yend
+        ),
+        colour = "#4e4d47",
+        linewidth = .2,
+        alpha = 0.8
+      ) +
+      geom_segment(
+        # the connecting segments between bars
+        data = df %>%
+          filter(group.id == group.id) %>%
+          group_by(group.id) %>%
+          slice_tail() %>%
+          ungroup() %>%
+          select(group.id, yend),
+        aes(
+          x = group.id,
+          xend = if_else(group.id == last(group.id), last(group.id), group.id + 1),
+          y = yend,
+          yend = yend
+        ),
+        linewidth = .3,
+        colour = "gray25",
+        linetype = "dashed",
+        show.legend = FALSE,
+        inherit.aes = FALSE
+      ) +
+      ggrepel::geom_text_repel(
+        # the labels
+        aes(
+          x = group.id,
+          label = paste0(scales::comma(abs(values))),
+          y = rowSums(cbind(ystart, values / 2)),
+          segment.colour = "gray33"
+        ),
+        nudge_x = -.5,
+        arrow = arrow(type = "open", length = unit(.1, "inches")),
+        colour = "#4e4d47",
+        # alpha = .7,
+        size = 5,
+        inherit.aes = FALSE
+      ) +
+      annotate(
+        geom = "text",
+        x = mid_plot,
+        y = max(df$yend) * 1.1,
+        size = 16/.pt,
+        label = paste0(
```

Comment:
This is great. I think we should turn the label into a function, since we'll likely reuse it in the other charts. I add this at the bottom of helper_functions.R:

```
sys_total_count_line <- function(total_count) {
  return(str_wrap(
    paste0(
      "Total ",
      case_when(
        input$syso_level_of_detail == "All" ~ "People",
        input$syso_level_of_detail == "HoHsOnly" ~ "Heads of Household",
        TRUE ~
          getNameByValue(syso_level_of_detail, input$syso_level_of_detail)
      ),
      case_when(
        input$syso_hh_type == "All" ~ "",
        str_detect(input$syso_hh_type, "Household") == FALSE ~
          paste(" in", getNameByValue(syso_hh_types, input$syso_hh_type),
                " Households"),
        TRUE ~
          paste(" in", getNameByValue(syso_hh_types, input$syso_hh_type))
      ),
      ": ",
      total_count
    ),
    width = 30
  ))
}
```

So you'd have:
```
annotate(
  geom = "text",
  x = mid_plot,
  y = max(df$yend) * 1.1,
  size = 16 / .pt,
  label = paste0(
    sys_total_count_line(total_clients),
    "\nTotal Change: ",
    inflow_to_outflow
  )
)
'''

Reply:
Done! Great idea, thank you @alex-silverman